File: fincore-frontend/src/types/transaction.ts
export interface Transaction {
  id: string;
  title: string;
  type: "Revenue" | "Expense";
  categoryId: string;
  categoryName: string;
  amount: number;
  transactionDate: string;
  description?: string;
  creatorName: string;
  status: "Pending" | "Approved" | "Rejected";
  approvedByName?: string;
  approvedAt?: string;
  rejectionReason?: string;
}

export interface CreateTransactionData {
  title: string;
  type: number; // 1 = Revenue, 2 = Expense
  categoryId: string;
  amount: number;
  transactionDate: string;
  description?: string;
}

export interface ApproveTransactionData {
  transactionId: string;
  status: number; // 2 = Approved, 3 = Rejected
  rejectionReason?: string;
}

export interface TransactionCategory {
  id: string;
  name: string;
  type: "Revenue" | "Expense";
}

export interface CreateCategoryData {
  name: string;
  type: number; // 1 = Revenue, 2 = Expense
}

File: fincore-backend/FinCore.Domain/Entities/Transaction.cs
using System;
using FinCore.Domain.Common;
using FinCore.Domain.Enums;

namespace FinCore.Domain.Entities;

public class Transaction : BaseEntity
{
    public string Title { get; set; } = null!;
    public TransactionType Type { get; set; }
    public Guid CategoryId { get; set; }
    public virtual TransactionCategory Category { get; set; } = null!;
    public decimal Amount { get; set; }
    public DateTime TransactionDate { get; set; }
    public string? Description { get; set; }
    public Guid CreatorId { get; set; }
    public virtual User Creator { get; set; } = null!;
    
    // Approval status details
    public TransactionStatus Status { get; set; } = TransactionStatus.Pending;
    public Guid? ApprovedById { get; set; }
    public virtual User? ApprovedBy { get; set; }
    public DateTime? ApprovedAt { get; set; }
    public string? RejectionReason { get; set; }
}
